src/OpenTelemetry.Api/Context/RuntimeContext.cs

```diff
@@ -49,9 +91,8 @@ public static RuntimeContextSlot<T> RegisterSlot<T>(string slotName)
                     throw new InvalidOperationException($"Context slot already registered: '{slotName}'");
                 }
 
-                var type = ContextSlotType.MakeGenericType(typeof(T));
-                var ctor = type.GetConstructor(new Type[] { typeof(string) });
-                var slot = (RuntimeContextSlot<T>)ctor.Invoke(new object[] { slotName });
+                var slot = runtimeContextSlotFactory.Create<T>(slotName);
```

Comment:
I think we could simplify the code by updating the definition of `RegisterSlot<T>` method like this:

```csharp
public static RuntimeContextSlot<T> RegisterSlot<T>(string slotName)
{
    Guard.ThrowIfNullOrEmpty(slotName);
    RuntimeContextSlot<T> slot;

    lock (Slots)
    {
        if (Slots.ContainsKey(slotName))
        {
            throw new InvalidOperationException($"Context slot already registered: '{slotName}'");
        }

        if (ContextSlotType == typeof(AsyncLocalRuntimeContextSlot<>))
        {
            slot = new AsyncLocalRuntimeContextSlot<T>(slotName);
        }
        else if (ContextSlotType == typeof(ThreadLocalRuntimeContextSlot<>))
        {
            slot = new ThreadLocalRuntimeContextSlot<T>(slotName);
        }
#if NETFRAMEWORK
        else if (ContextSlotType == typeof(RemotingRuntimeContextSlot<>))
            {
            slot = new RemotingRuntimeContextSlot<T>(slotName);
        }
#endif
        else
        {
            throw new NotSupportedException("${value} is not supported.");
        }

        Slots[slotName] = slot;
        return slot;
    }
}
```